CONTRIBUTING.md

```diff
@@ -0,0 +1,155 @@
+Ensure that the example is updated whenever the template itself is updated.
+
+## Making a project
+
+Each project is its own sub-directory in the `projects` root directory.
+The standard components of a project directory are:
+
+- `README.md` file.
+- `cookiecutter.json` file.
+- `{{cookiecutter.project_name}}` directory.
+- `example` directory.
+
+The next sections describe these files and directories in more detail.
+
+### Project README file
+
+The title for a project's `README.md` should match the project's directory name.
+
+Below the title, include a summary paragraph that describes what kinds of projects this project template can create.
+
+Include a section called `Template variables`.
+Subsections should match keys in the `cookiecutter.json` file and document each variable.
+
+Include a section called `Files`.
+Each subsection should be the name of a file (in a project cookiecutter directory).
+Here you can document each component of that file, and describe how developers should extend the file for their own project.
+
+### Project cookiecutter.json file
+
+The `cookiecutter.json` file is used by [Cookiecutter](https://cookiecutter.readthedocs.io/en/latest/) to get input from a user and create a project.
+
+The JSON file contains a single JSON object.
+Keys are names of Jinja templates and values are default values.
+For example:
+
+```json
+{
+  "copyright_year": 2017,
+  "copyright_holder": "Association of Universities for Research in Astronomy, Inc.",
+  "project_name": "my_project"
+}
+```
+
+Besides scalar default values, you can instead specify [choice variables](https://cookiecutter.readthedocs.io/en/latest/advanced/choice_variables.html) and [dictionary variables](https://cookiecutter.readthedocs.io/en/latest/advanced/dict_variables.html).
+
+See the [Cookiecutter documentation](https://cookiecutter.readthedocs.io/en/latest/), and other project templates, for more information and examples.
+
+### Project cookiecutter directory
+
+In a cookiecutter project template, both the paths of files and directories, and the files's content, are Jinja2 template variables.
+
+The root directory of a cookiecutter template is also a Jinja2 template.
+For example, if the  `cookiecutter.json` file has a field `project_name`, and the directory of the project should be that name, the root directory of the cookiecutter template is:
+
+```jinja
+{{cookiecutter.project_name}}
+```
+
+### Project example directory
+
+You can optionally include an example project.
+Unless projects of this type have meaningful naming schemas, the name of this directory can be `example`.
+
+You should create the `example` directory from cookiecutter command execution alone.
+Whenever the cookiecutter template is updated, the example should also be updated and committed.
+Ideally, example re-generation should be scripted (for example, with a `Makefile`), but we don't have a suggested pattern for this yet.
```

Comment:
Can't we just have simple SConstructs?  I know it isn't pretty, but it's what the rest of the stack does.